ops/internal/manage/staging.go

```diff
@@ -91,29 +108,27 @@ func InflateChainConfig(st *state.State, idx int) (*config.StagedChain, error) {
 	}
 
 	cfg.Roles = config.Roles{
-		SystemConfigOwner: config.NewChecksummedAddress(chainIntent.Roles.SystemConfigOwner),
-		ProxyAdminOwner:   config.NewChecksummedAddress(chainIntent.Roles.L1ProxyAdminOwner),
-		Guardian:          config.NewChecksummedAddress(st.AppliedIntent.SuperchainRoles.Guardian),
-		Proposer:          config.NewChecksummedAddress(chainIntent.Roles.Proposer),
-		UnsafeBlockSigner: config.NewChecksummedAddress(chainIntent.Roles.UnsafeBlockSigner),
-		BatchSubmitter:    config.NewChecksummedAddress(chainIntent.Roles.Batcher),
-		Challenger:        config.NewChecksummedAddress(chainIntent.Roles.Challenger),
+		SystemConfigOwner: config.NewChecksummedAddress(dc.FinalSystemOwner),
+		ProxyAdminOwner:   config.NewChecksummedAddress(dc.ProxyAdminOwner),
+		Guardian:          config.NewChecksummedAddress(dc.SuperchainConfigGuardian),
+		Proposer:          config.NewChecksummedAddress(dc.L2OutputOracleProposer),
+		UnsafeBlockSigner: config.NewChecksummedAddress(dc.P2PSequencerAddress),
+		BatchSubmitter:    config.NewChecksummedAddress(dc.BatchSenderAddress),
+		// Challenger:        config.NewChecksummedAddress(chainIntent.Roles.Challenger), // TODO
```

Comment:
Let's get these with opaque mapping queries / getters.

Reply:
Done in https://github.com/ethereum-optimism/superchain-registry/pull/1043/commits/0064ce7a22a3c400a7fb6db1f97db15eb53c1db3